src/index.js

```diff
@@ -4,40 +4,72 @@ import filter from 'feathers-query-filters';
 import errors from 'feathers-errors';
 import { select } from 'feathers-commons';
 import * as utils from './utils';
+import { Op } from 'sequelize';
 
 class Service {
   constructor (options) {
     if (!options) {
       throw new Error('Sequelize options have to be provided');
     }
 
-    if (!options.Model) {
-      throw new Error('You must provide a Sequelize Model');
+    if (!options.model) {
+      throw new Error('You must provide a Sequelize model name');
+    }
+
+    if (!options.master) {
+      throw new Error('You must provide a Sequelize master instance');
     }
 
     this.paginate = options.paginate || {};
-    this.Model = options.Model;
+    this.model = options.model;
+    this.master = options.master;
+    this.replicas = options.replicas;
     this.id = options.id || 'id';
     this.events = options.events;
     this.raw = options.raw !== false;
   }
 
-  getModel (params) {
-    return this.Model;
+  getModel (params, server) {
+    return server.model(this.model);
+  }
+
+  applyScope (params, server) {
+    let Model;
+
+    const sequelize = this.computeSequelizeParam(params.sequelize, server);
+
+    Model = this.getModel(params, server);
+
+    if (sequelize.scope) {
+      return Model.scope(sequelize.scope);
+    }
+
+    return Model;
   }
 
-  applyScope (params) {
-    if ((params.sequelize || {}).scope) {
-      return this.getModel(params).scope(params.sequelize.scope);
+  decideServer (method, params) {
+    if ((!params.forceMaster) && this.replicas && (this.replicas.length > 0) && (method === 'find' || method === 'get')) {
+      // Read queries go to replicas
+      return this.replicas[Math.floor(Math.random() * this.replicas.length)];
+    } else {
+      // Write queries go to master
+      return this.master;
     }
-    return this.getModel(params);
+  }
+
+  computeSequelizeParam (sequelize, server) {
+    if (typeof sequelize === 'function') {
+      return sequelize(server);
+    }
+    return sequelize || {};
   }
 
   extend (obj) {
     return Proto.extend(obj, this);
   }
 
   _find (params, getFilter = filter, paginate) {
+    const server = this.decideServer('find', params);
```

Comment:
We're calling this everywhere, and we're also calling `applyScope()` everywhere passing the server we got from this, so it might make sense to combine them into a single method?